lib/galaxy/tools/parameters/__init__.py

```diff
@@ -368,18 +363,8 @@ def populate_state(request_context, inputs, incoming, state, errors={}, context=
                 populate_state(request_context, input.inputs, incoming.get(input.name), group_state, errors, context=context, check=check, simple_errors=simple_errors, input_format='21.01')
 
             elif input.type == 'upload_dataset':
-                file_count = input.get_file_count(request_context, context)
-                while len(group_state) > file_count:
-                    del group_state[-1]
-                while file_count > len(group_state):
-                    new_state = {'__index__' : len(group_state)}
-                    for upload_item in input.inputs.values():
-                        new_state[upload_item.name] = upload_item.get_initial_value(request_context, context)
-                    group_state.append(new_state)
-                for i, rep_state in enumerate(group_state):
-                    rep_index = rep_state['__index__']
-                    rep_prefix = '%s_%d|' % (input.name, rep_index)
-                    populate_state(request_context, input.inputs, incoming.get(rep_prefix, []), rep_state, errors, context=context, check=check, simple_errors=simple_errors, input_format='21.01')
+                raise NotImplementedError
+
```

Comment:
Sorry, I would rather not work on this. The scope of this PR already got way bigger than I expected.